node/consensus/src/passthrough.rs

```diff
@@ -36,17 +36,22 @@ pub fn spawn_consensus(
                     },
                 };
                 acc.push(message);
-                future::ok((control_rx, acc))
+                Either::A(future::ok((control_rx, acc)))
             } else {
                 if !acc.is_empty() {
                     let c = ChainConsensusBlockBody { messages: acc };
-                    tokio::spawn(consensus_tx.clone().send(c).map(|_| ()).map_err(|e| {
-                        error!("Failure sending pass-through consensus {:?}", e);
-                    }));
+                    Either::B(consensus_tx.clone().send(c).then(|res| {
```

Comment:
What's the advantage of using `Either` here?

Reply:
It allows to combine futures of different types. If we remove Either compiler would barf because different places in the function return different types.